src/control/lib/control/network.go

```diff
@@ -225,6 +225,7 @@ type (
 		CrtCtxShareAddr uint32
 		CrtTimeout      uint32
 		NetDevClass     uint32
+		Version         uint32
```

Comment:
Seems a bit odd to me to note the version in the response to let the control plane do the version check. I would have expected the data plane to check whether the incoming request matches its version, and return some sort of error if not.

Reply:
There are two checks done by the agent.  The first check ensures that the DRPC_PROTOCOL_VERSION of the client matches the DRPC_PROTOCOL_VERSION that the agent was built with.  The second check ensures the the DRPC_PROTOCOL_VERSION that the agent was built with matches what the daos_server was built with.  Doing the server/agent check doesn't require data from the data plane per se.  But the agent won't have that information until it actually receives the server response from the GetAttachInfo.  If there is a need to do some additional version checking in the data plane itself, that can be added.  Though my understanding of the requirements for this story is that we're trying to make sure that the client <-> agent protocol is matched.  I added the agent <-> server protocol matching because it goes a step further to make sure that these three parts are in synchronization.  I could imagine doing additional checking in the IO servers, but I didn't think that was necessary here.

Reply:
I see what you mean.  I'll rework it so that the server checks the version in the request and does not pass back a version in the response.

Reply:
I've moved the server version check to the server side, and added unit tests.